lib/p5.fab.js

```diff
@@ -258,376 +269,463 @@ class Fab {
+    translate(this.radius, 0, this.radius);
+    rotateY(PI / 12);
+    rotateX(PI / 12);
+    fill(254, 249, 152);
+    push();
+    translate(0, 2.5, 0);
+    cylinder(this.radius + 1, 5); // build plate
+    pop();
+
+    push();
+    noFill();
+    translate(0, -this.maxZ / 2 + 1, 0);
+    stroke(220, 50, 32);
+    box((2 * this.radius) / sqrt(2), this.maxZ, (2 * this.radius) / sqrt(2)); // work envolope
+    pop();
+
+    // not sure if needed
+    noFill();
+    stroke(0);
+  }
+
+  /*****
+   * G-Code Commands
+   */
+  autoHome() {
+    var cmd = "G28";
+    this.add(cmd);
+    this.add("G92 E0");
+
+    return cmd;
+  }
+
+  setTemps(tNozzle, tBed) {
+    var cmd = `M104 S${tNozzle}`; // set nozzle temp without waiting
+    this.add(cmd);
+
+    cmd = `M140 S${tBed}`; // set bed temp without waiting
+    this.add(cmd);
+
+    // now wait for both
+    cmd = `M109 S${tNozzle}`;
+    this.add(cmd);
+    cmd = `M190 S${tBed}`;
+    this.add(cmd);
+
+    return cmd;
+  }
+
+  setNozzleTemp(t) {
+    var cmd = `M109 S${t}`; // wait for temp
+    this.add(cmd);
+    return cmd;
+  }
+
+  setBedTemp(t) {
+    var cmd = `M190 S${t}`;
+    this.add(cmd);
+    return cmd;
+  }
+
+  setAbsolutePosition() {
+    var cmd = "G90";
+    this.add(cmd);
+  }
+
+  setRelativePosition() {
+    var cmd = "G91";
+    this.add(cmd);
+  }
+
+  setERelative() {
+    var cmd = "M83";
+    this.add(cmd);
+  }
+
+  fanOn() {
+    var cmd = "M106";
+    this.add(cmd);
+  }
+
+  fanOff() {
+    var cmd = "M107";
+    this.add(cmd);
+  }
+
+  pausePrint(t = null) {
+    var cmd = t ? `M1 S${t}` : "M1 S10 this is a pause";
+    this.commandStream.unshift(cmd);
+  }
+
+  configure(config) {
+    this.coordinateSystem = config.coordinateSystem;
+    this.radius = config.radius;
+    this.nozzleR = config.nozzleDiameter / 2;
+    this.filamentR = config.filamentDiameter / 2;
+    this.baudRate = config.baudRate;
+    this.autoConnect = config.autoConnect;
+    this.maxZ = config.maxZ;
+    if (config.coordinateSystem == "delta") {
+      this.maxX = (2 * config.radius) / sqrt(2);
+      this.maxY = this.maxX;
+      this.centerX = 0;
+      this.centerY = 0;
+    } else {
+      this.maxX = config.maxX;
+      this.maxY = config.maxY;
+      this.centerX = config.maxX / 2;
+      this.centerY = config.maxY / 2;
+    }
+  }
+
+  stopPrint() {
+    this.commandStream = []; // clear commands
+    this.isPrinting = false;
+    // this.serial.close();
+    // this.serial.getPorts();
+    fabDraw();
+  }
+
+  restartPrinter() {
+    var cmd = "M999";
+    this.add(cmd);
+    this.print();
+  }
+
+  introLine(z = 0.3) {
+    if (this.coordinateSystem != "delta") {
+      this.move(0.1, 20, z, 85);
+      this.moveExtrude(0.1, 200, z, 25);
+      this.addComment("intro");
+      this.move(0.4, 200.0, z, 85);
+      this.moveExtrude(0.4, 20.0, z, 25);
+      this.addComment("intro");
+    } else {
+      for (let angle = 0; angle <= TWO_PI / 3; angle += TWO_PI / 50) {
```

Comment:
For delta printer the usual  intro line is a section of a circle. The code was adapted from your "2.5sketching" example 
<img width="227" alt="Screenshot 2023-02-21 at 14 58 47" src="https://user-images.githubusercontent.com/1855325/220380157-dba8e54d-23a5-4fb9-baf1-e3269e8b4a58.png">
<img width="258" alt="Screenshot 2023-02-21 at 14 58 59" src="https://user-images.githubusercontent.com/1855325/220380166-2e67353d-14b7-464d-af79-33686bb3ded4.png">